R/tree.R

```diff
@@ -268,6 +268,22 @@ is_printable_env <- function(x) {
     )
 }
 
+tree_as_list <- function(x){
+  # Make sure to get rid of attributes as well
+  # rlang::as_list() is depreciated
+  if (rlang::is_environment(x)) {
+    # Environments are funky as they dont have names before conversion to list
+    # but do after, so let them handle their conversion
+    return (as.list(x))
```

Comment:
This might need to be `as.list.environment()`, because if the environment has a class (like with R6 objects), `as.list()` will not dispatch to `as.list.environment()`.

Reply:
Also need to consider whether or not to set `all.names = TRUE`

Reply:
I had no idea the default was `FALSE` for that. Updated to `TRUE`. Since the purpose of `tree` is to show the true underlying structure of objects it seems only natural we would expose dot prefixed children.